lib/views/request/request_view.dart

```diff
@@ -155,19 +200,49 @@ class _RequestViewState extends State<RequestView> {
                       ],
                     ),
                   ),
+                  Expanded(
+                    flex: 2,
+                    child: Container(
+                      alignment: Alignment.center,
+                      child: Text(
+                        identifier == RadioButtonData.btcAddress
+                            ? 'BTC address'
+                            : 'Lightning invoice',
+                        style: const TextStyle(fontSize: 30),
+                      ),
+                    ),
+                  ),
                   Expanded(
                       flex: 2,
-                      child: QrImageView(
-                        data: invoice.bolt11,
-                        version: QrVersions.auto,
-                        backgroundColor: Colors.white,
+                      child: Container(
+                        alignment: Alignment.center,
+                        padding: const EdgeInsets.all(20),
+                        child: Text(
+                          identifier == RadioButtonData.btcAddress
+                              ? 'Receive payments at this address'
+                              : 'Receive payments via this lightning invoice',
+                          style: const TextStyle(fontSize: 20),
+                        ),
                       )),
+                  Expanded(
+                      flex: 4,
+                      child: btcAddress == null && invoice == null
+                          ? const CircularProgressIndicator()
+                          : QrImageView(
+                              data: invoice ?? btcAddress!,
+                              version: QrVersions.auto,
+                              backgroundColor: Colors.white,
+                            )),
+                  const Spacer(flex: 1),
                   Expanded(
                       flex: 1,
                       child: IconButton(
                           onPressed: () {
-                            Clipboard.setData(
-                                ClipboardData(text: invoice.bolt11));
+                            String text =
+                                identifier == RadioButtonData.btcAddress
+                                    ? btcAddress!
+                                    : invoice!;
```

Comment:
Can you use a separate method to handle this logic and then call `String text = getWhereToPay()`

this logic is always bad inside the UI and you have this in several places